tests/integration-tests/tests/ad_integration/test_ad_integration.py

```diff
@@ -255,7 +270,8 @@ def _directory_factory(
                 )
                 directory_stack_name = directory_stack.name
                 created_directory_stacks[region]["directory"] = directory_stack_name
-
+                if request.config.getoption("retain_ad_stack"):
+                    add_tag_to_stack(vpc_stack.name, "DO-NOT-DELETE", " ")
```

Comment:
Why do we need to update an existing stack with the tag?
I think this mechanism should be applied only to the brand new stack, not on existing ones.

Reply:
the vpc stack is created as a fixture, so I does not have the tag. So if an existing active directory is not found, then a new one is created and a DO-NOT-DELETE tag is added to the corresponding vpc stack.